app/controllers/application_controller.rb

```diff
@@ -1,2 +1,39 @@
 class ApplicationController < ActionController::API
+  private
+
+  set_current_tenant_through_filter
+  before_action :set_the_current_tenant
+
+  def set_the_current_tenant
+    return unless ENV["ENFORCE_TENANCY"]
+
+    tenant = Tenant.find_by(:external_tenant => user_identity)
```

Comment:
@syncrou Where would tenants get created when a user first connects to the catalog?

Reply:
I don't have an answer on that yet, though I can discuss with @gtanzillo.